### ISSUE TITLE
device-mapper: reload ioctl on veracrypt1 (253:3) failed: invalid argument Command failed.

### ISSUE THREAD

**Issue by rtins** [veracrypt/VeraCrypt#1010]
I am encountering a device-mapper error when attempting to mount a VeraCrypt container. Is

# Expected behavior

I am expecting to successfully mount a file 

# Observed behavior

I am receiving an error when attempting to mount the VeraCrypt container. The error message is: device-mapper: reload ioctl on veracrypt1 (253:3) failed: invalid argument Command failed. 

# Steps to reproduce

1. Open VeraCrypt (GUI)
2. Press Select File..
3. Select File
4. Press Mount

# Screenshots

# Your Environment

Please tell us more about your environment

**VeraCrypt version:** Veracrypt 1.25.9

**Operating system and version:** Linux Mint Cinnamon 21

**System type:**  64 bit

**Comment by rtins** [comment]
As john-larfour2 previously mentioned in issue #839 , when attempting to disable kernel crypto services, I am able to successfully mount the volume, but I encounter the error message `ls: cannot access 'mount/folder': Structure needs cleaning` on all folders/files except for "lost+found". Is there a solution to this issue?

**Comment by NastyaGrifon** [comment]
Same issue on VeraCrypt 1.25.9, Linux 6.5.9-1-MANJARO, unstable branch

**Comment by NastyaGrifon** [comment]
At the moment you can work around the error by using the `-m=nokernelcrypto` flag. Can only be used in CLI though

**Comment by thejeffmagill** [comment]
> At the moment you can work around the error by using the `-m=nokernelcrypto` flag. Can only be used in CLI though

But doesn't this force slower software encryption rather than using the kernel? I've read this same solution elsewhere, but doesn't really seem like a good solution, more of a stop gap.

**Comment by NastyaGrifon** [comment]
> force slower software encryption rather than using the kernel

You're right, using the `nokernelcrypto` flag can impact performance. I ran two benchmarks, one with the flag and one without. Here are the results:

<details> 
  <summary> Results without the flag</summary>

![2025-01-02_16-51](https://github.com/user-attachments/assets/07229d7f-8b0c-4ae3-b139-d76ecc4a1ac1)

</details>

<details> 
  <summary> Results with the flag</summary>

![2025-01-02_16-51_1](https://github.com/user-attachments/assets/b452ee9d-b8fc-4f3c-9c3c-dc5afce5894c)

</details>


As you can see, there _**is**_ a difference. The speed varies significantly for every cipher method. In some cases using crypto kernel actually degrades the performance. It is important to note that synthetic benchmarks do not represent the everyday use and is pretty much dependent on the hardware you're using.

While the flag might offer performance tradeoffs, it's important to understand the reported bug where software mounting fails. Even if users gain up to 30% speed, a failing mount renders the software unusable.
